### DISCUSSION TITLE
Error with node.write_value(value, varianttype): BadWriteNotSupported

### DISCUSSION THREAD

**Discussion by nkawerau** [FreeOpcUa/opcua-asyncio#1162] [Q&A] [answered]
Hi,

I am totally new to opc-ua. The last days I tried to set up an environment with an opcua-asyncio client for communication with a running opcua server on a PLC device. So far it worked quite well for my needs with basic functionalities. I can read and write node values with the provided command line tools ("uaread" and "uawrite") and the same is possible with the "opcua-client-gui" I've installed as well. When I use a Python script I can also read all node values from my opcua server with `value = await node.read_value()`.

But my problem is that it is not possible at all to write values with the client on the server, although the access level of the node is CurrentRead and CurrentWrite. I tried both `await node.write_value(new_value)` and  `await node.write_value(new_value, varianttype)`.

The corresponding error is:
BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
Traceback:
File "C:\Users\ESIADMIN\miniforge3\envs\opc_env\lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 565, in _run_script
    exec(code, module.__dict__)
File "C:\Users\ESIADMIN\PycharmProjects\opcProject\client-minimal2.py", line 36, in <module>
    asyncio.run(main())
File "C:\Users\ESIADMIN\miniforge3\envs\opc_env\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
File "C:\Users\ESIADMIN\miniforge3\envs\opc_env\lib\asyncio\base_events.py", line 646, in run_until_complete
    return future.result()
File "C:\Users\ESIADMIN\PycharmProjects\opcProject\client-minimal2.py", line 24, in main
    await node.write_value(True)
File "C:\Users\ESIADMIN\miniforge3\envs\opc_env\lib\site-packages\asyncua\common\node.py", line 235, in write_value
    await self.write_attribute(ua.AttributeIds.Value, dv)
File "C:\Users\ESIADMIN\miniforge3\envs\opc_env\lib\site-packages\asyncua\common\node.py", line 285, in write_attribute
    result[0].check()
File "C:\Users\ESIADMIN\miniforge3\envs\opc_env\lib\site-packages\asyncua\ua\uatypes.py", line 328, in check
    raise UaStatusCodeError(self.value)

Is there anything I can do to solve this problem on the client side or do I have to change settings on the server side so that it supports "writing the combination of value, status and timestamps"? I am puzzeld why it does not work since there are no problems writing values with the CLI tools and the "opcua-client-gui".

I would appreciate all kinds of hints and pieces of advice to solve this issue.

**Comment by AndreasHeine** [comment, marked as answer]
https://github.com/AndreasHeine/opcua-tutorial/blob/8f1dc98e8f2301514aff5fd83ea31fdb35c82029/client/write.py#L35

**Reply by nkawerau** [reply]
Thank you! It would be nice to have this example also here: https://opcua-asyncio.readthedocs.io/en/latest/usage/get-started/minimal-client.html
Why is it in the example of the Documentation not necessary to create the ua.DataValue object?

**Reply by AndreasHeine** [reply]
python is dynamically typed! opc ua is strict typed!
if you don't pass a DataValue or a Variant class asyncua tries to guess the type but that is not always correct... like python int -> can be opc ua int16 int32 int64 -> if it does not match the variables datatype the server tells you that!
